.travis.yml

```diff
@@ -5,47 +5,45 @@ node_js:
 services:
   - docker
 
+env:
+  global:
+    - BUILD_TIME="$(date '+%Y-%m-%d %H:%M:%S')"
+
 script:
   # Audit npm packages. Fail build whan a PR audit fails, otherwise report the vulnerability and proceed.
-  - if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then npx audit-ci --low -a node-fetch; else npx audit-ci --low -a node-fetch || true; fi
-  - npm run lint
-  - npm test
-  - npm run test:apollo:default
-  - npm run test:apollo:local
-  - npm run test:apollo:passport.local
-  - export BUILD_TIME=`date '+%Y-%m-%d %H:%M:%S'`
-  - docker build --rm --build-arg BUILD_TIME="$BUILD_TIME" --build-arg BUILD_ID="${TRAVIS_BUILD_NUMBER}-${TRAVIS_COMMIT}" -t "quay.io/razee/razeedash-api:${TRAVIS_COMMIT}" .
+  - travis_fold start "npx.audit-ci" && if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then npx audit-ci; else npx audit-ci || true; fi && travis_fold end "npx.audit-ci"
+  - travis_fold start "npm.lint" && npm run lint && travis_fold end "npm.lint"
+  - travis_fold start "npm.test" && npm test && travis_fold end "npm.test"
+  - travis_fold start "npm.test:apollo:default" && npm run test:apollo:default && travis_fold end "npm.test:apollo:default"
+  - travis_fold start "npm.test:apollo:local" && npm run test:apollo:local && travis_fold end "npm.test:apollo:local"
+  - travis_fold start "npm.test:apollo:passport.local" && npm run test:apollo:passport.local && travis_fold end "npm.test:apollo:passport.local"
+  - travis_fold start "docker.build" && docker build --rm --build-arg BUILD_TIME="$BUILD_TIME" --build-arg BUILD_ID="${TRAVIS_BUILD_NUMBER}-${TRAVIS_COMMIT}" -t "quay.io/razee/razeedash-api:${TRAVIS_COMMIT}" . && travis_fold end "docker.build"
   - if [ -n "${TRAVIS_TAG}" ]; then docker tag quay.io/razee/razeedash-api:${TRAVIS_COMMIT} quay.io/razee/razeedash-api:${TRAVIS_TAG}; fi
   - docker images
   - ./build/process-template.sh kubernetes/razeedash-api/resource.yaml >/tmp/resource.yaml
+  - if [[ "${TRAVIS_TAG}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-[0-9]+)?$ ]]; then npm version --no-git-tag-version "${TRAVIS_TAG}"; fi
 
 before_deploy:
   - docker login -u="${QUAY_ID}" -p="${QUAY_TOKEN}" quay.io
 
 deploy:
```

Comment:
if i understand correctly...
- on release candidates builds (1.2.3-1): 
  - docker push
  - add file to release assets
  - publish to npm
- on full release builds (1.2.3):
  - add file to release assets

does that sound right? do you not want npm or docker pushes for some reason?

Reply:
ah, you consolidated the pushes into one regex. cool.

Reply:
yes